### PR TITLE
fix: Use monitoredApplicationWithProcessIdentifier API to fetch application objects from PID

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Related to https://github.com/appium/WebDriverAgent/pull/774

### CODE REVIEW

**Review by KazuCocoa — APPROVED** [review]
No change... but this pr itself lgtm.

File: WebDriverAgentLib/Utilities/FBXCodeCompatibility.m [hunk 1]
```diff
@@ -23,15 +23,6 @@
 
 @implementation XCUIApplication (FBCompatibility)
 
-+ (instancetype)fb_applicationWithPID:(pid_t)processID
-{
-  if (0 == processID) {
```
Comment (Dan-Maor) [inline WebDriverAgentLib/Utilities/FBXCodeCompatibility.m:28]: Are you sure this protection is not needed? What happens if the requested pid is 0?

**Comment by mykola-mokhnach** [comment]
> No change... but this pr itself lgtm.

Maybe then try to add logs after each line in the fb_activeApplicationWithDefaultBundleId method to precisely isolate the particular API call that creates the delay. Then we may try to find alternatives to it or find out what exactly creates the delay

**Comment by mykola-mokhnach** [comment]
I am not 100% convinced to merge this PR especially taking into account it does not fix the original issue related to testmanagerd crashes.

**Pull request closed without merging**